javagen/src/main/java/com/azure/autorest/util/ClientModelUtil.java

```diff
@@ -149,6 +125,28 @@ public static void getAsyncSyncClients(Client client, ServiceClient serviceClien
         }
     }
 
+    private static List<ConvenienceMethod> getConvenienceMethods(Supplier<List<ClientMethod>> clientMethods, OperationGroup og) {
+        return og.getOperations().stream()
+                .filter(o -> o.getConvenienceApi() != null)
+                .flatMap(o -> {
+                    List<ClientMethod> cMethods = Mappers.getClientMethodMapper().map(o, false)
+                            .stream()
+                            .filter(m -> m.getMethodVisibility() == JavaVisibility.Public)
+                            .collect(Collectors.toList());
+                    if (!cMethods.isEmpty()) {
+                        // sync stack generates additional proxy methods with name suffix "Sync"
+                        Set<String> proxyMethodNames = cMethods.stream()
+                                .map(m -> m.getProxyMethod().getName())
+                                .collect(Collectors.toSet());
+                        return clientMethods.get().stream()
+                                .filter(m -> proxyMethodNames.contains(m.getProxyMethod().getName()) && m.getMethodVisibility() == JavaVisibility.Public)
```

Comment:
I am not sure "contains" is best solution. Probably more specific about "Sync"?

E.g. client may have "list" and "listByFoo", or something like that.

Reply:
Got it. Replaced with proxyMethod.getBaseName(). Thanks!